osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs

```diff
@@ -139,6 +149,17 @@ protected override void Update()
             // This is basically allowing for correct alignment as relative pieces move around them.
             rightArea.Padding = new MarginPadding { Left = leftArea.DrawWidth };
             hitTargetOffsetContent.Padding = new MarginPadding { Left = HitTarget.DrawWidth / 2 };
+
+            // When rewinding, make sure to remove any auxilliary hit notes that were
+            // spawned and played during a drumroll.
+            if (Time.Elapsed < 0)
+            {
+                foreach (var o in drumRollHitContainer.Objects)
```

Comment:
this is going to cause a lag spike on the first negative frame of a very long beatmap.

Reply:
actually not even that, is going to add a constant iteration overhead.  we might want to do this in alive objects instead..

Reply:
It's also very unsafe since it's doing a removal inside a `foreach`, leading to collection enumeration exceptions.

Alive objects could work.

Reply:
Alrighty. Fair enough.

What exactly are alive objects? How should I go about improving this?

Reply:
i'm already making the change